controllers/apidefinition_controller.go

```diff
@@ -542,7 +542,9 @@ func (r *ApiDefinitionReconciler) delete(ctx context.Context, desired *tykv1alph
 		r.Log.Info("deleting api")
 
 		_, err = klient.Universal.Api().Delete(ctx, desired.Status.ApiID)
-		if err != nil {
+		if err != nil && err.Error() == "Resource not found" {
```

Comment:
Could you use this function to check if error is `Resource not found` https://github.com/TykTechnologies/tyk-operator/blob/c4b85072dce3a66eae6dce59e1d7eb20090f8a86/pkg/client/client.go#L38

Reply:
That was my first thought, but I wasn't sure about importing that package only for the error check. I'll go ahead and use `client.IsNotFound`